sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/listeners/OnMultipleRequestListener.java

```diff
@@ -0,0 +1,58 @@
+package com.smartdevicelink.proxy.rpc.listeners;
+
+import android.util.Log;
+
+import com.smartdevicelink.proxy.RPCResponse;
+import com.smartdevicelink.proxy.rpc.enums.Result;
+
+import org.json.JSONException;
+
+import java.util.Vector;
+
+/**
+ * This is the listener for sending Multiple RPCs.
+ */
+public abstract class OnMultipleRequestListener extends OnRPCResponseListener {
+
+	Vector<Integer> correlationIds;
+	OnRPCResponseListener rpcResponseListener;
+	private static String TAG = "OnMultipleRequestListener";
+
+	public OnMultipleRequestListener(){
+		setListenerType(UPDATE_LISTENER_TYPE_MULTIPLE_REQUESTS);
+		if(correlationIds == null){
+			correlationIds = new Vector<>();
+		}
+		rpcResponseListener = new OnRPCResponseListener() {
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				correlationIds.remove(Integer.valueOf(correlationId));
+				if(correlationIds.size()>0){
+					onUpdate(correlationIds.size());
+				}else{
+					onFinished();
+				}
+			}
+		};
+	}
+
+	public void addCorrelationId(int correlationid){
+		if(correlationIds == null){
```

Comment:
The `correlationIds` should never be null if set it to final so this block can be removed.